src/main/java/de/tum/cit/aet/artemis/core/config/EurekaClientConfiguration.java

```diff
@@ -17,16 +14,18 @@
 import org.springframework.cloud.netflix.eureka.http.RestClientDiscoveryClientOptionalArgs;
 import org.springframework.cloud.netflix.eureka.http.RestClientTransportClientFactories;
 import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Conditional;
 import org.springframework.context.annotation.Configuration;
-import org.springframework.context.annotation.Profile;
 import org.springframework.web.client.RestClient;
 
+import de.tum.cit.aet.artemis.core.config.conditions.BuildAgentOrCoreCondition;
+
 /**
  * This class is necessary to avoid using Jersey (which has an issue deserializing Eureka responses) after the spring boot upgrade.
  * It provides the RestClientTransportClientFactories and RestClientDiscoveryClientOptionalArgs that would normally not be instantiated
  * when Jersey is found by Eureka.
  */
-@Profile({ PROFILE_CORE, PROFILE_BUILDAGENT })
+@Conditional(BuildAgentOrCoreCondition.class)
```

Comment:
This is an interesting case: If we want to define more complex conditions based on profile AND/OR properties, we can do that with nested properties like `BuildAgentOrCoreCondition `